src/main/java/org/redhat/appstudio/serviceprovider/service/storage/VaultAccessTokenService.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2021 Red Hat, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.redhat.appstudio.serviceprovider.service.storage;
+
+import io.quarkus.vault.VaultKVSecretEngine;
+import io.quarkus.vault.runtime.client.VaultClientException;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.jboss.logging.Logger;
+
+public class VaultAccessTokenService implements AccessTokenService {
+
+  private static final Logger LOG = Logger.getLogger(VaultAccessTokenService.class);
```

Comment:
It would be nice to see an integration test with that. Do you want to have it as part of this or next pr?